informer/src/handler/connection.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Handler, Sink, PubSubHandler, Session, MetaIoHandler, Params, Error, ErrorCode, SubscriptionId, Subscriber, Value, PubSubMetadata};
+use jsonrpc_core::{futures, BoxFuture};
+use futures::Future;
+use std::sync::Arc;
+use std::{thread, time};
+
+
+pub struct Connection{
+    pub status: bool,
+    pub intrested_event: Vec<String>,
+    pub sink: Sink,
+}
+
+impl Connection{
+    pub fn new(status: bool, events: String , sink: Sink) -> Self {
+
+        Self{
+            status: status,
+            intrested_event: vec![events.clone()],
+            sink: sink.clone(),
+        }
+    }
+    pub fn event_subscription <T> (&mut self, mut io : PubSubHandler<T>) {
```

Comment:
Will you add this function in the Connection struct? I feel awkward. The connection will be created whenever a user is connected. `io.add_subscription` will be called only once.

Reply:
I think `io.add_subscription` will be called whenever a user sends a subscription request. and will not be called once.

Reply:
Since their documentation is not good, it is hard to know their intention.
With my common sense about webserver, what `io.add_subscription` is doing is registering a callback to the WebSocket server. In web server, we do not register callback multiple times.

In my assumption, the closure you passed to the `io.add_subscription`' will be called whenever user connects.

Could you check the `io.add_subscription`'s behavior?

Reply:
`add_subscription` creates a new `subscribe` and `unsubscribe` RPC methods which their types are `SubscribeRpcMethod` and `UnsubscribeRpcMethod`. Then add these two so-called asynchronous methods to RPC request handler (`MetaIoHandler`). Therefore, I cannot conclude whether we need to create a subscription and unsubscription for each event by ourself (before anybody sends any subscription request) or we need to create a subscription/un per each request.